lib/DocumentManager.php

```diff
@@ -11,9 +11,26 @@
 
 namespace Sulu\Component\DocumentManager;
 
+use Sulu\Component\DocumentManager\Exception\DocumentManagerException;
 use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
+/**
+ * A document manager has two roles:.
+ *
+ * - Provider of an API for working with documents.
+ * - A "sub container", providing access to the services which fall
+ *   within the scope of the particular document manager (as there can be many).
+ *
+ * This class is therefore an event dispatcher and a service container and has
+ * no other business logic within it.
+ *
+ * All primary methods are wrapped in a try catch block, the exception is then
+ * wrapped in a `DocumentManagerException` with a reference to this document
+ * managers name.
+ *
+ * TODO: Create a proxy document manager which automatically wraps all calls.
```

Comment:
So this proxy document manager would implement the `_call` method, or how would you implement that?